test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
         # See reservation in degreaded state
         a = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
         d = self.server.expect(RESV, a, rid)
+
+    @timeout(400)
+    def test_resv_jobend(self):
+        """
+        Test if a node goes back to free state when reservation ends and
+        admin-suspended job is killed
+        """
+
+        # Submit a reservation
+        r = Reservation(TEST_USER)
+        r.set_attributes({'Resource_List.select': '1:ncpus=1',
+                          'reserve_start': time.time() + 60,
+                          'reserve_end': time.time() + 120})
+        rid = self.server.submit(r)
+
+        # See reservation is confirmed
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        d = self.server.expect(RESV, a, rid)
+
+        # Submit a job
+        j = Job(TEST_USER)
+        rque = rid.split(".")
+        j.set_attributes({'queue': rque[0]})
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Wait for reservation to start
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|3')}
+        d = self.server.expect(RESV, a, rid, offset=60, max_attempts=120)
+
+        # job is running as well
+        self.server.expect(
+            JOB, {'job_state': 'R', 'substate': 42}, attrop=PTL_AND,
+            id=jid, max_attempts=30)
+
+        # Admin-suspend job
+        self.server.sigjob(jid, 'admin-suspend', runas=ROOT_USER)
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+        # Submit another job outside of reservation
+        j = Job(TEST_USER)
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        # Wait for the reservation to get over
+        # Job also gets deleted and node state goes back to free
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=120)
+        self.server.expect(NODE, {'state': 'free'}, id='vn[0]')
+
+        # job2 starts running
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2, max_attempts=60)
+
+    def test_que(self):
+        """
+        Test to check that job gets suspended on non-default queue
+        """
+
+        # create a high priority workq2 and a routeq
+        a = {'queue_type': 'execution', 'started': 't', 'enabled': 't',
+             'priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='workq2')
+        a = {'queue_type': 'route', 'started': 't', 'enabled': 't',
+             'route_destinations': 'workq2'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='route')
```

Comment:
Just trying to imitate with the real user scenario. You are right but there is no harm having it there.